.gitignore

```diff
@@ -0,0 +1,160 @@
+# Byte-compiled / optimized / DLL files
+__pycache__/
+*.py[cod]
+*$py.class
+
+# C extensions
+*.so
+
+# Distribution / packaging
+.Python
+build/
+develop-eggs/
+dist/
+downloads/
+eggs/
+.eggs/
+lib/
+lib64/
+parts/
+sdist/
+var/
+wheels/
+share/python-wheels/
+*.egg-info/
+.installed.cfg
+*.egg
+MANIFEST
+
+# PyInstaller
+#  Usually these files are written by a python script from a template
+#  before PyInstaller builds the exe, so as to inject date/other infos into it.
+*.manifest
+*.spec
+
+# Installer logs
+pip-log.txt
+pip-delete-this-directory.txt
+
+# Unit test / coverage reports
+htmlcov/
+.tox/
+.nox/
+.coverage
+.coverage.*
+.cache
+nosetests.xml
+coverage.xml
+*.cover
+*.py,cover
+.hypothesis/
+.pytest_cache/
+cover/
+
+# Translations
+*.mo
+*.pot
+
+# Django stuff:
+*.log
+local_settings.py
+db.sqlite3
+db.sqlite3-journal
+
+# Flask stuff:
+instance/
+.webassets-cache
+
+# Scrapy stuff:
+.scrapy
+
+# Sphinx documentation
+docs/_build/
+
+# PyBuilder
+.pybuilder/
+target/
+
+# Jupyter Notebook
+.ipynb_checkpoints
+
+# IPython
+profile_default/
+ipython_config.py
+
+# pyenv
+#   For a library or package, you might want to ignore these files since the code is
+#   intended to run in multiple environments; otherwise, check them in:
+# .python-version
```

Comment:
Is this supposed to be commented out? I would assume it is related to the code comment above so should this be uncommented? Same comments for pipenv and poetry.

Reply:
Thanks for this comment. This file comes directly from the GitHub [Python.gitignore](https://github.com/github/gitignore/blob/main/Python.gitignore). [Pyenv](https://github.com/pyenv/pyenv) can be used to set local environments per project, which creates a `.python-version` file. I generally don't use these myself, and wouldn't want to impose a specific Python version on development through this file as we're supporting several versions of Python.

For Poetry, my feeling is that we always want to include a `poetry.lock` file to help navigate dependencies together as a project team. This follows the `.gitignore`'s recommendation as well:

> Similar to Pipfile.lock, it is generally recommended to include poetry.lock in version control.

Reply:
I'll remove the commented portions to help clean this file up.